### DISCUSSION TITLE
Use `act` in Gitlab?

### DISCUSSION THREAD

**Discussion by guettli** [nektos/act#1922] [General]
Migrating between Github and Gitlab can be difficult, if you have a lot of CI configuration.

Just a thought/idea: 

Can `act` help here, so that you have a portable CI?

Can I run `act` in Gitlab?

**Comment by ChristopherHX** [comment]
`act` has far better compatibilty to https://github.com/go-gitea/gitea (very similar rest and url structure to GitHub) than to Gitlab (different repo urls and own rest api).

> Can I run `act` in Gitlab?

You will notice that all actions using any rest api will fail in gitlab and more will be broken.

I have never tried to do that, but I expect you have to rewrite a lot of actions to work together with GitLab. Maybe you even have to tweak act to be able to use Gitlab hosted Actions...
